db/db_impl/db_impl.cc

```diff
@@ -3992,6 +3996,28 @@ ArenaWrappedDBIter* DBImpl::NewIteratorImpl(
   return db_iter;
 }
 
+std::unique_ptr<MultiCfIterator> DBImpl::NewMultiCfIterator(
+    const ReadOptions& _read_options,
+    const std::vector<ColumnFamilyHandle*>& column_families) {
+  assert(column_families.size() > 0);
+  const Comparator* first_comparator = column_families[0]->GetComparator();
+
+  for (size_t i = 1; i < column_families.size(); ++i) {
+    if (first_comparator->GetId().compare(
```

Comment:
Please let me know if there's a better way to do this.

Reply:
For efficiency, you probably want to try pointer == first, which should be a common case.